chainio/clients/txsender/privatekey_txsender.go

```diff
@@ -0,0 +1,149 @@
+package txsender
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eth"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/signerv2"
+	sdktypes "github.com/Layr-Labs/eigensdk-go/types"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+var (
+	FallbackGasTipCap = big.NewInt(15_000_000_000)
+)
+
+var _ TxSender = (*privateKeyTxSender)(nil)
+
+type privateKeyTxSender struct {
+	ethClient eth.Client
+	address   common.Address
+	signerFn  signerv2.SignerFn
+	logger    logging.Logger
+
+	// cache
+	contracts map[common.Address]*bind.BoundContract
+}
+
+func NewPrivateKeyTxSender(ethClient eth.Client, chainID *big.Int, ecdsaPrivateKey *ecdsa.PrivateKey, logger logging.Logger) (TxSender, error) {
```

Comment:
SignerV2 should be input to this function. not ecdsa private key. since if we have web3 signer we won't have access to it. we should create signer before calling it based on SignerConfig and pass here. or you are thinking we would define `NewKeyStoreTxSender` and `NewWeb3TxSender`?

Reply:
these should be Signers which we have and txSender constructor should just accept signerV2 itself

Reply:
I was thinking we'd have separate txsender implementations for each signer and that's why I called it `PrivateKeyTxSender` because it uses the `PrivateKeySignerFn`. But I don't see why we need separate txsender per signer method, so I'll consolidate it and make it take SignerV2 as an input.

Reply:
yep. I think interface should be same since it's same behavior of transaction sending which is the TxSender/Wallet's job. It should not have knowledge of signer for this.